FreeRTOS-Plus/Demo/AWS/Ota_Windows_Simulator/Ota_Over_Mqtt_Demo/DemoTasks/OtaOverMqttDemoExample.c

```diff
@@ -0,0 +1,2146 @@
+
+/**
+ * @brief Buffer used to store the code signing certificate file path.
+ * Buffer is passed to the OTA agent during initialization.
+ */
+static uint8_t certFilePath[ otaexampleMAX_FILE_PATH_SIZE ];
+
+/**
+ * @brief Buffer used to store the name of the data stream.
+ * Buffer is passed to the OTA agent during initialization.
+ */
+static uint8_t streamName[ otaexampleMAX_STREAM_NAME_SIZE ];
+
+/**
+ * @brief Buffer used decode the CBOR message from the MQTT payload.
+ * Buffer is passed to the OTA agent during initialization.
+ */
+static uint8_t decodeMem[ ( 1U << otaconfigLOG2_FILE_BLOCK_SIZE ) ];
+
+/**
+ * @brief Application buffer used to store the bitmap for requesting firmware image
+ * chunks from MQTT broker. Buffer is passed to the OTA agent during initialization.
+ */
+static uint8_t bitmap[ OTA_MAX_BLOCK_BITMAP_SIZE ];
+
+/**
+ * @brief A statically allocated array of event buffers used by the OTA agent.
+ * Maximum number of buffers are determined by how many chunks are requested
+ * by OTA agent at a time along with an extra buffer to handle control message.
+ * The size of each buffer is determined by the maximum size of firmware image
+ * chunk, and other metadata send along with the chunk.
+ */
+static OtaEventData_t eventBuffer[ otaconfigMAX_NUM_OTA_DATA_BUFFERS ] = { 0 };
+
+/*
+ * @brief Mutex used to manage thread safe access of OTA event buffers.
+ */
+static SemaphoreHandle_t xBufferSemaphore;
+
+/**
+ * @brief Static handle used for MQTT agent context.
+ */
+static MQTTAgentContext_t xGlobalMqttAgentContext;
+
+/*---------------------------------------------------------*/
+
+/**
+ * @brief Task for MQTT agent.
+ * Task runs MQTT agent command loop, which returns only when the user disconnects
+ * MQTT, terminates agent, or the mqtt connection is broken. If the mqtt connection is broken, the task
+ * suspends OTA agent reconnects to the broker and then resumes OTA agent.
+ *
+ * @param[in] pParam Can be used to pass down functionality to the agent task
+ */
+static void prvMQTTAgentTask(void* pParam);
+
+/**
+ * @brief Function used by OTA agent to publish control messages to the MQTT broker.
+ *
+ * The implementation uses MQTT agent to queue a publish request. It then waits
+ * for the request complete notification from the agent. The notification along with result of the
+ * operation is sent back to the caller task using xTaksNotify API. For publishes involving QOS 1 and
+ * QOS2 the operation is complete once an acknwoledgment (PUBACK) is received. OTA agent uses this function
+ * to fetch new job, provide status update and send other control related messges to the MQTT broker.
+ *
+ * @param[in] pacTopic Topic to publish the control packet to.
+ * @param[in] topicLen Length of the topic string.
+ * @param[in] pMsg Message to publish.
+ * @param[in] msgSize Size of the message to publish.
+ * @param[in] qos Qos for the publish.
+ * @return OtaMqttSuccess if successful. Appropirate error code otherwise.
```

Comment:
```suggestion
 * @return OtaMqttSuccess if successful. Appropriate error code otherwise.
```

Please do a find and replace in this file too.